### ISSUE TITLE
CA2208 should allow property name for property setters instead of 'value'

### ISSUE THREAD

**Issue by pharring** [dotnet/roslyn-analyzers#1561]
#### Analyzer package

Microsoft.NetCore.Analyzers, version [2.6.0](https://www.nuget.org/packages/Microsoft.NetCore.Analyzers/2.6.0)

#### Analyzer

CA2208: Instantiate argument exceptions correctly

#### Repro steps

Analyze this code:

```C#
using System;

namespace ClassLibrary1
{
    public class Class1
    {
        private int _size;
        public int Size
        {
            get => _size;
            set
            {
                if (value < 0)
                {
                    throw new ArgumentOutOfRangeException(nameof(Size));
                }

                _size = value;
            }
        }
    }
}
```

#### Expected behavior

No code analysis warnings.
I think the rule should be relaxed to allow one to use the name of the property here instead of "value".

#### Actual behavior

CA2208:	Method set_Size passes 'Size' as the paramName argument to a ArgumentOutOfRangeException constructor. Replace this argument with one of the method's parameter names. Note that the provided parameter name should have the exact casing as declared on the method.

**Comment by sharwell** [comment]
I would be in favor of treating the current implementation as By Design on this one.

**Comment by mavasani** [comment]
I am going to close this out as By Design, and relax the implementation if we get further feedback about the noise level.

**Comment by mavasani** [comment]
We ave another report of this issue here: https://developercommunity.visualstudio.com/content/problem/480715/vs2019-rosyln-analyzers-dont-work-correctly-in-pro.html

**Comment by SledgeHammer01** [comment]
This should be fixed. In a project with extensive property validation, the noise is excessive.

**Comment by mavasani** [comment]
I don't see real need to flag such a case, the fix should also be pretty trivial. Just needs tweaking https://github.com/dotnet/roslyn-analyzers/blob/75064eeea4bc85beae38c7f8bdde6a839739edb5/src/Microsoft.NetCore.Analyzers/Core/Runtime/InstantiateArgumentExceptionsCorrectly.cs#L102
